lms/static/js/edxnotes/plugins/llm_summarize.js

```diff
@@ -68,16 +68,107 @@
+                editor.fields[1].element.children[0].value = 'ai_summary';
+                toggleLoader(editor);
+
+                const request = new Request('/pearson-core/llm-assistance/api/v0/summarize-text', {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'X-CSRFToken': $.cookie('csrftoken'),
+                    },
+                    body: JSON.stringify({
+                        text_to_summarize: annotation.quote,
+                    }),
+                });
+                fetch(request)
+                    .then((response) => {
+                        toggleLoader();
+                        if (response.ok) return response.json();
+                        throw new Error(gettext('There was an error while summarizing the content.'));
+                    })
+                    .then((data) => {
+                        textAreaWrapper.children[0].value = data.summary;
+                    })
+                    .catch((error) => {
+                        alert(error.message);
+                        editor.hide();
+                    });
+
+                function toggleLoader() {
+                    const saveButton = document.querySelector('.annotator-controls .annotator-save');
+                    const loaderWrapper = document.querySelector('.summarize-loader-wrapper');
+                    editor.fields[0].element.children[0].classList.toggle('d-none');
+                    loaderWrapper.classList.toggle('d-none');
+                    saveButton.disabled = !saveButton.disabled;
+                }
+            },
+            cleanupSummarize: function(editor) {
+                const textAreaWrapper = editor.fields[0].element;
+                textAreaWrapper.children[0].value = '';
+                textAreaWrapper.children[1].value = '';
+                editor.element[0].setAttribute('is_summarizing', 'false');
+                const loaderWrapper = document.querySelector('.summarize-loader-wrapper');
+                loaderWrapper.classList.add('d-none');
+            },
+            modifyDom: function(annotator) {
                 annotator.adder[0].children[0].id = 'annotateButton';
                 annotator.adder[0].children[0].innerHTML = '<i class="fa fa-pencil" aria-hidden="true"></i>';
                 annotator.adder[0].innerHTML += `
                 <button class="summarize-button" id="summarizeButton" title="${gettext('Summarize text using AI.')}">
                     <i class="fa fa-star" aria-hidden="true"></i>
                 </button>
                 `;
+                const textAreaWrapper = annotator.editor.fields[0].element;
+                textAreaWrapper.innerHTML += `
+                <div class="summarize-loader-wrapper d-none">
+                    <span class="loader" style="
+                        // Style is being defined here since classes styling not working
```

Comment:
Have you used id or class selector? @alexjmpb Please, try to be more specific on the selector, because if this is the case, how are the others styles working?

Reply:
Yes, even using id it's not applying the styles, I also tried with bs and fa spinners.